cookbook/x-flat-jwt.md

```diff
@@ -0,0 +1,324 @@
+HTTP/1.1 200 OK
+…
+Content-Type: application/json
+
+{
+  "args": {},
+  "data": "",
+  "files": {},
+  "form": {},
+  "headers": {
+    "Accept": "*/*",
+    "Accept-Encoding": "deflate, gzip",
+    "Host": "httpbin.org",
+    "User-Agent": "curl/7.29.0",
+    "X-Amzn-Trace-Id": "…" 
+  },
+  "json": null,
+  "method": "GET",
+  "origin": "…",
+  "url": "https://httpbin.org/anything" 
+}
+```
+
+That's what you would expect from httpbin.org, right?
+
+## Restricting access: Swagger Security and `x-flat-jwt`
+
+Now, you don't want anyone except authorized users to use this proxy. This is typically achieved with access tokens. Some tokens are JSON Web Tokens ([JWT](https://tools.ietf.org/html/rfc7519)), while others are opaque.
+
+Swagger has a two-part feature to describe protected access to routes: `securityDefinitions` (what sort of protection is applied …) and `security` (… to which routes), e.g.:
+
+```yaml
+swagger: "2.0" 
+basePath: /
+securityDefinitions:
+  JWTCookie:
+    type: apiKey
+    in: header
+    name: Cookie
+paths:
+  /httpbin/**:
+    security:
+      - JWTCookie: []
+```
+
+This defines a security scheme object (named `JWTCookie`), meaning that some sort of cookie is needed to access certain routes. This is applied to the [wildcard path](/reference/OpenAPI/differences.md#wildcard-paths) `/httpbin/**`.
+
+This documentation feature, with some extensions, is used to make FLAT actually permit access to the route only if a valid JWT token is presented.
+
+First, we define the name of the cookie expected to accompany the API request:
+
+```yaml
+…
+securityDefinitions:
+  JWTCookie:
+    type: apiKey
+    in: header
+    name: Cookie
+    x-flat-cookiename: authtoken    # ⬅ specify the cookie name
+…
+```
+
+Then we specify the [configuration](/reference/OpenAPI/security.md#the-x-flat-jwt-field) for decoding the JWT token:
+
+```yaml
+…
+  JWTCookie:
+    type: apiKey
+    in: header
+    name: Cookie
+    x-flat-cookiename: authtoken
+    x-flat-jwt:                     # ⬅ our JWT configuration:
+      key:                          # ⬅ the key to decode the JWT …
+        file: secret.pem            # ⬅ … is read from the file secret.pem
+      alg: RS256                    # ⬅ the signing algorithm is RS256
+…
+```
+
+The specified key is a public key, read from the file secret.pem, e.g.:
+```
+-----BEGIN PUBLIC KEY-----
+MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDGSd+sSTss2uOuVJKpumpFAaml
+t1CWLMTAZNAabF71Ur0P6u833RhAIjXDSA/QeVitzvqvCZpNtbOJVegaREqLMJqv
+FOUkFdLNRP3f9XjYFFvubo09tcjX6oGEREKDqLG2MfZ2Z8LVzuJc6SwZMgVFk/63
+rdAOci3W9u3zOSGj4QIDAQAB
+-----END PUBLIC KEY-----
+```
+
+That's all. Now FLAT will only permit requests if they supply a token that bear an RS256 signature that was created with the private key that matches the given public key.
+
+(You may want to try this with the key and algorithm provided to you by the party providing you with tokens, e.g. an OAuth2 authorization server.)
```

Comment:
provided … providing

Reply:
wie waer's mit `the key and algo acquired from the party ...`?